src/grid/grid_enum.h

```diff
@@ -69,7 +69,29 @@ enum : int {
   num_fields,
 
 //Aliases and manually computed enums
+  #ifdef  MHD
```

Comment:
This looks good! 

Adding more comments 

- Wrap the magnetic fluxes in ifdef MHD so that it compiles for non-MHD builds 
- You have the option of doing something like
```
  fluxX_magnetic_z = magnetic_start,
  fluxX_magnetic_y, // instead of  fluxX_magnetic_y = magnetic_start+1,
```
but this is a case where I imagine you're preferring explicit over implicit
- I have a slight preference for moving ```nscalars = finalscalar_plus_1 - scalar,``` to the top of the list since everything else is being wrapped by ifdef MHD but NSCALARS always needs to be defined. 



```
// Aliases and manually computed enums
  nscalars = finalscalar_plus_1 - scalar, // Prefer this to be at the top of list or at the end
  
  #ifdef  MHD
  num_flux_fields = num_fields-1,
  num_interface_fields = num_fields-1,
  #else
  num_flux_fields = num_fields,
  num_interface_fields = num_fields,
  #endif  //MHD

  #ifdef MHD
  magnetic_start = magnetic_x,
  magnetic_end   = magnetic_z, // Unused, consider 
  // Note that the direction of the flux, the suffix _? indicates the direction of the electric field, not the magnetic flux
  fluxX_magnetic_z = magnetic_start,
  fluxX_magnetic_y = magnetic_start+1,
  fluxY_magnetic_x = magnetic_start,
  fluxY_magnetic_z = magnetic_start+1,
  fluxZ_magnetic_y = magnetic_start,
  fluxZ_magnetic_x = magnetic_start+1,
  Q_x_magnetic_y = magnetic_start,
  Q_x_magnetic_z = magnetic_start+1,
  Q_y_magnetic_z = magnetic_start,
  Q_y_magnetic_x = magnetic_start+1,
  Q_z_magnetic_x = magnetic_start,
  Q_z_magnetic_y = magnetic_start+1
  #endif
```

Reply:
- I'm in the middle of fixing it for non-mhd builds now
- I'd prefer explicit. Also, IMO everything in the "aliases" section should be explicit since that's the section where we're saying ordering doesn't matter
- I moved `nscalars = finalscalar_plus_1 - scalar,`

The commit for all of this should be up shortly

Reply:
If you like you can also move these over. This is just the output of ```git grep -ip +[\s]*NSCALARS```

Just wanted to point it out but no worries if it's not a priority. 

```
src/mhd/ct_electric_fields_tests.cu=public:
src/mhd/ct_electric_fields_tests.cu:        grid (n_cells * (8+NSCALARS)),
src/mhd/magnetic_update_tests.cu=public:
src/mhd/magnetic_update_tests.cu:        sourceGrid      (n_cells * (8+NSCALARS)),
src/mhd/magnetic_update_tests.cu:        destinationGrid (n_cells * (8+NSCALARS), -999.),
src/mhd/magnetic_update_tests.cu:        fiducialData    (n_cells * (8+NSCALARS), -999.),
src/utils/mhd_utilities_tests.cu=TEST(tMHDCellCenteredMagneticFields,
src/utils/mhd_utilities_tests.cu:    std::vector<double> testGrid(n_cells * (8+NSCALARS));
```

Reply:
Those are all tests and aren't actually indexing, they're creating a mock grid for testing. I've replaced them with `grid_enum::num_fields` and it appears to be working. There's a few other uses floating around in the tests, and only there as far as I can tell, and I replaced them with `grid_enum::nscalars` since they're actually counting scalars and not indexing for the magnetic field

Reply:
I also didn't touch any of the scalar loops in PCM or the HLLD solver. I don't know enough about scalars to risk breaking something.

Reply:
Raw NSCALARS (no plus signs) can be left as is or replaced, they are being macro-replaced with grid_enum::nscalars anyway. grid_enum is designed to maintain compatibility with the scalar loops throughout the code by correctly calculating NSCALARS.

Reply:
Sounds good. I'm going to mark this conversation as resolved, feel free to reopen it if needed.